spring-web/src/main/java/org/springframework/web/util/UriComponentsBuilder.java

```diff
@@ -764,6 +765,15 @@ public UriComponentsBuilder replaceQueryParam(String name, Object... values) {
 		return this;
 	}
 
+	@Override
+	public UriComponentsBuilder replaceQueryParam(String name, @Nullable Object value) {
+		Collection<?> values = null;
+		if (!ObjectUtils.isEmpty(value)) {
```

Comment:
`value` is not an array type, so it seems strange to check with `ObjectUtils.isEmpty()`.

Reply:
`value` doesn't need to be an array. Please check the definition of `ObjectUtils.isEmpty()` here: https://github.com/gdagitrep/spring-framework/blob/621295dbd8e53dd66901348e6f23fbdfd78b8b22/spring-core/src/main/java/org/springframework/util/ObjectUtils.java#L134

Reply:
Oh, I didn't realize if there were other functions with different parameters. As the number of overloaded functions increases, the convenience of using the function increases, but the readability of the code seems to decrease.

Reply:
Agreed

Reply:
@cuspymd Possible to approve if no concerns?